### PR TITLE
Enforce trailing peer limits when select pivot block fails

### PR DESCRIPTION
**Pull request by frankisawesome**
Signed-off-by: Frank Li <b439988l@gmail.com>

## PR description
Rather than waiting 5 seconds and hoping everything goes well the next run we should run the trailing peer limiter task and drop the useless peers.
## Fixed Issue(s)

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java [hunk 1]
```diff
@@ -69,6 +72,13 @@ public FastSyncActions(
     this.ethContext = ethContext;
     this.syncState = syncState;
     this.metricsSystem = metricsSystem;
+    this.trailingPeerLimiter =
+        new TrailingPeerLimiter(
+            ethContext.getEthPeers(),
+            () ->
+                new TrailingPeerRequirements(
+                    syncState.getLocalChainHeight() + syncConfig.getFastSyncPivotDistance(),
```
Comment (matkt) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java:80]: are we sure about that ? If I'm close to the head (<getFastSyncPivotDistance)

Comment (frankisawesome) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java]: I changed it to use the min of local + pivot distance or the current best estimate, will do some testing today

**Comment by frankisawesome** [comment]
I've set up two test instances on goerli and the sync finished in 4h8m and 4h15m respectively, quite happy with this.

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java [hunk 2]
```diff
@@ -177,10 +179,26 @@ private CompletableFuture<FastSyncState> retrySelectPivotBlockAfterDelay() {
     return ethContext
         .getScheduler()
         .scheduleFutureTask(
+            this::limitTrailingPeersAndRetrySelectPivotBlock, Duration.ofSeconds(5));
+  }
+
+  private long conservatitvelyEstimatedPivotBlock() {
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java:185]: s/conservatitvely/conservatively

**Review by macfarla — APPROVED** [review]
LGTM. added a comment re a typo

**Pull request merged**